packages/rocketchat-oembed/server/server.js

```diff
@@ -0,0 +1,301 @@
+			return stream.abort();
+		}
+	});
+	stream.on('data', function(chunk) {
+		chunks.push(chunk);
+		chunksTotalLength += chunk.length;
+		if (chunksTotalLength > 250000) {
+			return stream.abort();
+		}
+	});
+	stream.on('end', Meteor.bindEnvironment(function() {
+		if (error != null) {
+			return callback(null, {
+				error,
+				parsedUrl
+			});
+		}
+		const buffer = Buffer.concat(chunks);
+		return callback(null, {
+			headers,
+			body: toUtf8(headers['content-type'], buffer),
+			parsedUrl,
+			statusCode
+		});
+	}));
+	return stream.on('error', function(err) {
+		return error = err;
+	});
+};
+
+OEmbed.getUrlMeta = function(url, withFragment) {
+	const getUrlContentSync = Meteor.wrapAsync(getUrlContent);
+	const urlObj = URL.parse(url);
+	if (withFragment != null) {
+		const queryStringObj = querystring.parse(urlObj.query);
+		queryStringObj._escaped_fragment_ = '';
+		urlObj.query = querystring.stringify(queryStringObj);
+		let path = urlObj.pathname;
+		if (urlObj.query != null) {
+			path += `?${ urlObj.query }`;
+		}
+		urlObj.path = path;
+	}
+	const content = getUrlContentSync(urlObj, 5);
+	if (!content) {
+		return;
+	}
+	if (content.attachments != null) {
+		return content;
+	}
+	let metas = undefined;
+	if (content && content.body) {
+		metas = {};
+		content.body.replace(/<title[^>]*>([^<]*)<\/title>/gmi, function(meta, title) {
+			return metas.pageTitle != null ? metas.pageTitle : metas.pageTitle = he.unescape(title);
+		});
+		content.body.replace(/<meta[^>]*(?:name|property)=[']([^']*)['][^>]*\scontent=[']([^']*)['][^>]*>/gmi, function(meta, name, value) {
+			let name1;
+			return metas[name1 = changeCase.camelCase(name)] != null ? metas[name1] : metas[name1] = he.unescape(value);
+		});
+		content.body.replace(/<meta[^>]*(?:name|property)=["]([^"]*)["][^>]*\scontent=["]([^"]*)["][^>]*>/gmi, function(meta, name, value) {
+			let name1;
+			return metas[name1 = changeCase.camelCase(name)] != null ? metas[name1] : metas[name1] = he.unescape(value);
+		});
+		content.body.replace(/<meta[^>]*\scontent=[']([^']*)['][^>]*(?:name|property)=[']([^']*)['][^>]*>/gmi, function(meta, value, name) {
+			let name1;
+			return metas[name1 = changeCase.camelCase(name)] != null ? metas[name1] : metas[name1] = he.unescape(value);
+		});
+		content.body.replace(/<meta[^>]*\scontent=["]([^"]*)["][^>]*(?:name|property)=["]([^"]*)["][^>]*>/gmi, function(meta, value, name) {
+			let name1;
+			return metas[name1 = changeCase.camelCase(name)] != null ? metas[name1] : metas[name1] = he.unescape(value);
+		});
+		if (metas.fragment === '!' && (withFragment == null)) {
+			return OEmbed.getUrlMeta(url, true);
+		}
+	}
+	let headers = undefined;
+
+
+	if (content && content.headers) {
+		headers = {};
+		const headerObj = content.headers;
+		Object.keys(headerObj).forEach((header) => {
+			headers[changeCase.camelCase(header)] = headerObj[header];
+		});
+	}
+	if (content && content.statusCode !== 200) {
+		return;
```

Comment:
@rodrigok But here `data` would just return undefined since the variable is being created just after that return, or did i miss something here?